pkg/common/cns-lib/volume/manager.go

```diff
@@ -155,8 +155,12 @@ func (m *volumeManager) CreateVolume(spec *cnstypes.CnsVolumeCreateSpec) (*cnsty
 			klog.Errorf("Failed to get taskInfo for CreateVolume task from vCenter %q with err: %v", m.virtualCenter.Config.Host, err)
 			return nil, err
 		}
-		// Store the taskInfo details and taskInfo object expiration time in volumeTaskMap
-		volumeTaskMap[spec.Name] = createVolumeTaskDetails{taskInfo, time.Now().Add(time.Hour * time.Duration(defaultOpsExpirationTimeInHours))}
+		// There is no need add task details into volumeTaskMap for static volume provisioning
+		// as it does not result in orphaned volumes. This is handled by checking if BackingDiskId is present in CreateVolume Spec
+		if spec.BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails).BackingDiskId == "" {
```

Comment:
I think we need to handle Null pointer access here. In the case when spec.BackingObjectDetails is nil, accessing BackingDiskId from nil reference will result into panic.

Reply:
Good catch! Done